### PR TITLE
#120 Fix schema alignment check

### LINKED ISSUE #120: Fix typo in schema alignment
## Describe the bug
https://github.com/AbsaOSS/hermes/blob/develop/datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparator.scala#L169

add `!`

### PR DESCRIPTION
**Pull request by Zejnilovic**
### Description
- add new spark-commons and update commons
- fix the schema alignment check, bad brackets

### Previous behaviour
When using passing schema do dataset comparison, it would say schema is bad even though it wasn't

### Current behaviour
Passing the schema to dataset comparison works again

### Modules affected
- [x] DatasetComparison
- [ ] InfoFileComparison
- [ ] E2ERunner

### Other
- [ ] Has new configs 
- [ ] Requires Docs update
- [ ] Introduces new logs
- [ ] Introduces new error messages

Fixes #120

### CODE REVIEW

**Review by dk1844 — DISMISSED** [review]
LGTM (just read the code)

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparator.scala [hunk 1]
```diff
@@ -165,10 +165,9 @@ class DatasetComparator(dataFrameReference: DataFrame,
   def checkSchemas(testedDF: ComparisonPair[DataFrame], schema: StructType): Unit = {
     val expectedSchema: StructType = getSchemaWithoutMetadata(testedDF.reference.schema)
     val actualSchema: StructType = getSchemaWithoutMetadata(testedDF.actual.schema)
-
-    if (!(SchemaUtils.isSubset(schema, actualSchema) && SchemaUtils.isSubset(schema, expectedSchema))) {
-      val diffSchema = SchemaUtils.diffSchema(schema, actualSchema) ++
-        SchemaUtils.diffSchema(schema, expectedSchema)
+    if (!schema.isSubset(actualSchema) && !schema.isSubset(expectedSchema)) {
```
Comment (benedeki) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparator.scala]: Why not to use `isEquivalent` as above?

Comment (Zejnilovic) [reply datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparator.scala]: This is checking if the `schema` is a subset of 2 different schemas; `actualSchema` and `expectedSchema`. No equivalency check going on here.

Comment (benedeki) [reply datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparator.scala]: Ah, see. Sorry

File: datasetComparison/src/test/scala/za/co/absa/hermes/datasetComparison/DatasetComparatorSuite.scala [hunk 2]
```diff
@@ -23,6 +23,7 @@ import za.co.absa.hermes.datasetComparison.cliUtils.CliParameters
 import za.co.absa.hermes.datasetComparison.config.ManualConfig
 import za.co.absa.hermes.datasetComparison.dataFrame.{Parameters, Utils}
 import za.co.absa.hermes.utils.SparkTestBase
+import za.co.absa.spark.commons.implicits.StructTypeImplicits.StructTypeEnhancements
```
Comment (AdrianOlosutean) [inline datasetComparison/src/test/scala/za/co/absa/hermes/datasetComparison/DatasetComparatorSuite.scala]: Probably unused?

**Review by AdrianOlosutean — DISMISSED** [review]

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparator.scala [hunk 3]
```diff
@@ -165,10 +165,9 @@ class DatasetComparator(dataFrameReference: DataFrame,
   def checkSchemas(testedDF: ComparisonPair[DataFrame], schema: StructType): Unit = {
     val expectedSchema: StructType = getSchemaWithoutMetadata(testedDF.reference.schema)
     val actualSchema: StructType = getSchemaWithoutMetadata(testedDF.actual.schema)
-
-    if (!(SchemaUtils.isSubset(schema, actualSchema) && SchemaUtils.isSubset(schema, expectedSchema))) {
-      val diffSchema = SchemaUtils.diffSchema(schema, actualSchema) ++
-        SchemaUtils.diffSchema(schema, expectedSchema)
+    if (!schema.isSubset(actualSchema) || !schema.isSubset(expectedSchema)) {
```
Comment (Zejnilovic) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparator.scala:168]: Made it easier

**Review by benedeki — APPROVED** [review]

**Review by benedeki — APPROVED** [review]

**Review by dk1844 — APPROVED** [review]

**Pull request merged**